test/config/utility.cc

```diff
@@ -114,6 +114,13 @@ void ConfigHelper::finalize(const std::vector<uint32_t>& ports) {
     }
   }
   ASSERT(port_idx == ports.size());
+
+#ifdef __APPLE__
+  // Under heavy load (and in particular in CI), macOS connections can take inordinately long to
+  // complete.
+  setConnectTimeout(std::chrono::seconds(30));
```

Comment:
Thought: rather than doing this, you may want to setConnectionTimeout(default timeout we have today) and then in the set function scale up for apple.

reason being if someone wants to set a higher or lower timeout this may not behave as you'd expect for Apple.
I'm totally fine with you sticking with this and revising if needed since you pretty much singlehandedly support the macos builds and can always tweak later if you run into the thing I'm concerned about :-)

Reply:
Let me poke at it a bit.